src/theme/components/socialmedialinks.ts

```diff
@@ -0,0 +1,168 @@
+// maybe set the border to always be on, but invisible if prop is set to "none"?
+
+import { StyleFunctionProps } from "@chakra-ui/theme-tools";
+
+function getRadius(size) {
+  let radiusSize = "18px";
+  switch (size) {
+    case "medium":
+      radiusSize = "24px";
+      break;
+    case "large":
+      radiusSize = "32px";
+  }
+  return {
+    radiusSize,
+  };
+}
+
+function getColor(color) {
+  let ltColor = "ui.typography.heading",
+    dkColor = "dark.ui.typography.heading",
+    ltBgColor,
+    dkBgColor;
+
+  switch (color) {
+    case "link":
+      ltColor = "ui.link.primary";
+      dkColor = "dark.ui.link.primary";
+      break;
+    case "textInverse":
+      ltColor = "ui.typography.inverse.heading";
+      dkColor = "dark.ui.typography.inverse.heading";
+      ltBgColor = "ui.typography.heading";
+      dkBgColor = "dark.ui.typography.heading";
+  }
+
+  return {
+    ltColor,
+    dkColor,
+    ltBgColor,
+    dkBgColor,
+  };
+}
+
+const SocialMediaLinks = {
+  // The parts of the component
+  parts: ["li", "a", "svg", "span"],
+  // The base styles for each part
+  baseStyle: {
+    // ul styles are not specifically identified as ul:
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "flex-start",
+    gap: "xs",
+    li: {
+      padding: "xs",
+    },
+    a: {
+      display: "flex",
+      alignItems: "center",
+      gap: "xs",
+      alignSelf: "stretch",
+      fontStyle: "normal",
+      fontSize: "desktop.body.body2",
+      textDecoration: "none",
+      color: "ui.typography.heading",
+      _dark: {
+        color: "dark.ui.typography.heading",
+      },
+    },
+    svg: {
+      fill: "ui.typography.heading",
+      _dark: {
+        fill: "dark.ui.typography.heading",
+      },
+    },
+  },
+  // The variants are bordered
+  variants: {
+    // Each variant is a function that returns stuff
+    straight(props: StyleFunctionProps) {
+      const { color, layout } = props;
+      let theseColors = getColor(color);
+      return {
+        flexDirection: layout,
+        li: {
+          borderRadius: "button.default",
+          borderWidth: "button.default",
```

Comment:
The border is not showing up and I think it's because this value should be "1px" instead of "button.default".